cloudinit/distros/alpine.py

```diff
@@ -202,6 +205,374 @@ def preferred_ntp_clients(self):
+            use_shadow = True
+
+        # If Shadow's 'passwd' is available then use the generic
+        # expire_passwd function from __init__.py instead.
+        if use_shadow:
+            return super().expire_passwd(user)
+
+        # Busybox's 'passwd' does not provide an expire option
+        # so have to manipulate the shadow file directly.
+        shadow_contents = None
+        shadow_file = self.shadow_fn
+        try:
+            shadow_contents = util.load_text_file(shadow_file)
+        except FileNotFoundError as e:
+            util.logexc(
+                LOG, "Failed to read %s file, file not found", shadow_file
+            )
+            raise e
+
+        # Find the line in /etc/shadow for the user
+        original_line = None
+        for line in shadow_contents.splitlines():
+            current_user = line.split(":")[0]
+            if current_user == user:
+                LOG.debug("Found /etc/shadow line matching user %s", user)
+                original_line = line
+
+        if original_line is not None:
+            # Replace existing line for user with modified line
+            new_list = original_line.split(":")
+            # Field '2' (numbered from '0') in /etc/shadow
+            # is the "date of last password change".
+            if new_list[2] != "0":
+                # Busybox's 'adduser' always expires password so only
+                # need to expire it now if this is not a new user.
+                new_list[2] = "0"
+                new_line = ":".join(new_list)
+                shadow_contents = shadow_contents.replace(
+                    original_line, new_line, 1
+                )
+
+                LOG.debug("Expiring password for user %s", user)
+                try:
+                    util.write_file(
+                        shadow_file,
+                        shadow_contents,
+                        omode="w",
+                        preserve_mode=True,
+                    )
+                except IOError as e:
+                    util.logexc(LOG, "Failed to update %s file", shadow_file)
+                    raise e
+            else:
+                LOG.debug("Password for user %s is already expired", user)
+        else:
+            util.logexc(LOG, "Failed to set 'expire' for %s", user)
+
+    def create_group(self, name, members=None):
+        # If 'groupadd' is available then use the generic
+        # create_group function from __init__.py instead.
+        if subp.which("groupadd"):
+            return super().create_group(name, members)
+
+        group_add_cmd = ["addgroup", name]
+        if not members:
+            members = []
+
+        # Check if group exists, and then add if it doesn't
+        if util.is_group(name):
+            LOG.warning("Skipping creation of existing group '%s'", name)
+        else:
+            try:
+                subp.subp(group_add_cmd)
+                LOG.info("Created new group %s", name)
+            except Exception:
```

Comment:
```suggestion
            except subp.ProcessExecutionException:
```

Reply:
Done.

Reply:
Actually pylint doesn't like this change:

```
cloudinit/distros/alpine.py:528: [E1101(no-member), Distro.create_group] Module 'cloudinit.subp' has no 'ProcessExecutionException' member
```

Reply:
Bah, sorry. I tried (and failed) from memory. It should be ProcessExecutionError

Reply:
Fixed.